src/agent/src/mount.rs

```diff
@@ -778,11 +778,18 @@ pub async fn add_storages(
             }
         };
 
-        // Todo need to rollback the mounted storage if err met.
-        let mount_point = res?;
-
-        if !mount_point.is_empty() {
-            mount_list.push(mount_point);
+        match res {
+            Ok(mount_point) => {
+                if !mount_point.is_empty() {
+                    mount_list.push(mount_point);
+                }
+            }
+            Err(e) => {
+                for m in mount_list {
+                    let _ = umount(m.as_str());
```

Comment:
It seems fine to go on with the rollback if umount() fails but the failure should be logged anyway.